src/NSwag.Generation.AspNetCore/AspNetCoreOpenApiDocumentGenerator.cs

```diff
@@ -404,18 +404,32 @@ private bool RunOperationProcessors(OpenApiDocument document, ApiDescription api
             return true;
         }
 
-        private string GetOperationId(OpenApiDocument document, ControllerActionDescriptor actionDescriptor, MethodInfo method)
+        private string GetOperationId(OpenApiDocument document, ControllerActionDescriptor actionDescriptor, MethodInfo method, string httpMethod)
         {
             string operationId;
 
             dynamic swaggerOperationAttribute = method
                 .GetCustomAttributes()
                 .FirstAssignableToTypeNameOrDefault("SwaggerOperationAttribute", TypeNameStyle.Name);
 
+            dynamic httpAttribute = null;
+            if (!string.IsNullOrWhiteSpace(httpMethod))
+            {
+                var attributeName = Char.ToUpperInvariant(httpMethod[0]) + httpMethod.Substring(1).ToLowerInvariant();
+                var typeName = string.Format("Microsoft.AspNetCore.Mvc.Http{0}Attribute", attributeName);
+                httpAttribute = method
+                    .GetCustomAttributes()
+                    .FirstAssignableToTypeNameOrDefault(typeName);
+            }
+
             if (swaggerOperationAttribute != null && !string.IsNullOrEmpty(swaggerOperationAttribute.OperationId))
             {
                 operationId = swaggerOperationAttribute.OperationId;
             }
+            else if (httpAttribute != null && !string.IsNullOrWhiteSpace(httpAttribute.Name))
```

Comment:
Maybe we should add a setting, eg Settings.UseHttpAttributeNameAsOperationId (better name?, default false) to not break existing users and make it less opinionated?

Reply:
I added a setting, please see latest commits